pkg/api/rest/update.go

```diff
@@ -103,3 +104,57 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx api.Context, obj, old runtime
 
 	return nil
 }
+
+type TransformFunc func(ctx api.Context, newObj runtime.Object, oldObj runtime.Object) (runtime.Object, error)
+
+type defaultUpdatedObjectInfo struct {
+	obj          runtime.Object
+	transformers []TransformFunc
+}
+
+func DefaultUpdatedObjectInfo(obj runtime.Object, transformers ...TransformFunc) UpdatedObjectInfo {
+	return &defaultUpdatedObjectInfo{obj, transformers}
+}
+
+func (i *defaultUpdatedObjectInfo) Preconditions() *api.Preconditions {
+	preconditions := &api.Preconditions{}
+
+	// Attempt to get the UID out of the object
+	accessor, err := meta.Accessor(i.obj)
+	if err != nil {
+		return preconditions
+	}
+
+	// If non-empty, set as a precondition
+	uid := accessor.GetUID()
+	if len(uid) > 0 {
+		preconditions.UID = &uid
+	}
+
+	return preconditions
+}
+
+func (i *defaultUpdatedObjectInfo) UpdatedObject(ctx api.Context, oldObj runtime.Object) (runtime.Object, error) {
+	var err error
+	// Start with the configured object
+	newObj := i.obj
+	if newObj != nil {
+		// If the original is non-nil (might be nil if the first transformer builds the object from the oldObj), make a copy,
+		// so we don't return the original. BeforeUpdate can mutate the returned object, doing things like clearing ResourceVersion.
+		// If we're re-called, we need to be able to return the pristine version.
+		newObj, err = api.Scheme.Copy(newObj)
```

Comment:
I'm ok with either forcing people to pass in a Copier or having the caller do it.  Just not using Scheme directly (rest handler should also not see scheme directly).